app.js

```diff
@@ -293,54 +278,77 @@ function handleMongoConnectionState(err) {
             return process.exit();
         }
 
-        console.log('Starting app.');
+        console.log('starting app');
         startApp();
     });
 
 }
 
 const Kefir = require('kefir');
 const interval = 2000;
+const Promise = require('bluebird');
+const retry = require('bluebird-retry')
 const MongoClient = require('mongodb').MongoClient;
+const chalk = require('chalk');
+const debug = require('debug')('app');
 
 function tryConnect(callback) {
-    console.log('Connecting to database...');
-    console.log(new Date());
-    console.log(`connection is ${settings.database.uri}`)
-    MongoClient.connect(settings.database.uri, callback);
 
+    console.log(chalk.green('Connecting to database...'));
+    console.log(chalk.green(new Date()));
+    console.log(chalk.green(`connection is ${settings.database.uri}`));
+
+    let p, wrapper = {}
+
+try{
+
+    p= MongoClient.connect(settings.database.uri,
+       {connectTimeoutMS:3000, reconnectTries:1, promiseLibrary:Promise});
+       console.log('!!');
+
+    wrapper = new Promise((resolve, reject)=>{
+         p.then(resolve, reject);
+    });
 
     mongoose.connection.on('error', function (err) {
-        console.log('connection on error ${err}');
-        //callback(err);
+        console.log(chalk.red(`connection on error ${err}`));
+        callback(err);
     });
 
     mongoose.connection.on('disconnected', function (err) {
         console.log(`disconnected from database error : ${err}`);
-          //callback("error");
+          callback("error");
     });
-    //if (retryNum < 3)
-    //  throw new Error('number of connection bigger the 3');
+}catch(e){
+  console.log(e + exception);
+  return Promise.reject();
+}
+process.on('uncaughtException', function (err) {
+  debug('uncaughtException: probably due to ' + err);
+})
+
+console.log(chalk.green('connection in progress'));
+p.then(()=>{
+  chalk.green('promise resolved ');
+},()=>{
+  chalk.green('promise rejected ');
+})
+return p;
+
+}
+let checkTimeout = ()=>{
+  console.log('check timeout');
+  return new Promise((resolve, reject)=>{
+    setTimeout(1000, ()=>{
+      console.log('rejected')
+      return reject();
+    })
+  })
 }
 
-  let startStream = Kefir.repeat((e)=>{
-    if (e>5) return false;
-    console.log(`e = ${e}`);
-    return Kefir.later(1000, e);
-  }).flatMap((retries)=>{
-    console.log(`retries ${retries}`)
-    if (retries<process.env.MOGNO_CONNECT_ATTEPMTS||3)
-    return Kefir.fromNodeCallback(tryConnect)
-  });
-
-  //startStream.onError((e)=>{throw e});
-
-  let everythingOk = startStream.ignoreErrors()
-  .take(1)
-  .onValue(handleMongoConnectionState);
-
-startStream.onEnd(()=>{
-  console.log('cant connect to DB after few retries');
-  process.exit(1);
+  let p = retry(tryConnect, { max_tries: 5, interval: 4000,timeout:30000 })
```

Comment:
If you'd wish to drop `bluebird-retry`, you could achieve the same effect using vanilla js:
```javascript
let connectToDatabase = ()=> {
	return MongoClient.connect(settings.database.uri, { connectTimeoutMS:3000, reconnectTries:1 });
};

// Tap into your function's Promise-rejections and exception throws, and coerce a Promise return
let tryPromise = (func)=> {
	return new Promise((resolve, reject)=> {
		let result = _.attempt(func);
		_.isError(result) ? reject(result) : result.then(resolve, reject);
	});
};

// Repeat x times
function repeatPromise(func, attempts = 10, delay = 1000){
	let attemptNumber = 0;  
  return (function internalTry(){
    attemptNumber++;
    if(attemptNumber > attempts) { 
	  	return Promise.reject('Exceeded number of attempts');
	  } else {
	    console.log(`Trying ${attemptNumber}`);
	  	return tryPromise(func).catch(()=> new Promise((resolve)=> _.delay(resolve, delay)).then(internalTry));  
  	}
  })();
};

// Try to connect to database
repeatPromise(connectToDatabase, 15, 1000)
	.then(console.log)
    .catch(console.warn);
```
`tryPromise` and `repeatPromise` would probably be useful in additional contexts as well, so it might be a good idea to add them to a central promise function lib (if such exists).